rxandroid/src/main/java/io/reactivex/android/schedulers/AndroidSchedulers.java

```diff
@@ -43,8 +44,31 @@ public static Scheduler mainThread() {
 
     /** A {@link Scheduler} which executes actions on {@code looper}. */
     public static Scheduler from(Looper looper) {
+        return from(looper, false);
+    }
+
+    /**
+     * A {@link Scheduler} which executes actions on {@code looper}.
+     *
+     * @param async if true, the scheduler will use async messaging on API >= 16 to avoid VSYNC
+     *              locking. On API < 16, this will no-op.
+     * @see android.os.Message#setAsynchronous(boolean)
+     */
+    @SuppressLint("NewApi")
+    public static Scheduler from(Looper looper, boolean async) {
         if (looper == null) throw new NullPointerException("looper == null");
-        return new HandlerScheduler(new Handler(looper));
+        boolean useAsync = async && Build.VERSION.SDK_INT >= 16;
+        // Confirm the method is there
+        Handler handler = new Handler(looper);
+        if (useAsync) {
+            Message message = Message.obtain(handler);
+            try {
+                message.setAsynchronous(true);
+            } catch (NoSuchMethodError e) {
+                useAsync = false;
+            }
+        }
+        return new HandlerScheduler(new Handler(looper), useAsync);
```

Comment:
You already created the `Handler` above. Can pass it here.

Reply:
derp, 6fbdff3